### DISCUSSION TITLE
Does the message preview feature risk getting me IP banned

### DISCUSSION THREAD

**Discussion by warzer13** [ScratchAddons/ScratchAddons#5039] [Help] [answered]
When you use this extension, it has the ability to give you previews of your scratch inbox. Does it allow scratch to view your IP address when you aren't on the scratch website?

**Comment by CST1229** [comment, marked as answer]
I'm pretty sure Scratch does not ban using browser extensions, only mentioning them on the website.

**Reply by warzer13** [reply]
No, can you please read my question properly? Thank you.

**Reply by WorldLanguages** [reply]
@warzer13 Your question is very weird. But the answer for the question in the body (not in the title) is obviously "yes".

**Reply by warzer13** [reply]
Whoops, sorry. My bad. I should have named it "does the message preview feature risk getting me IP banned". Overall its a great extension and I would use it nonetheless.

**Reply by warzer13** [reply]
Thanks for the answer.

**Reply by warzer13** [reply]
Does it still give the scratch team your IP if you use a different chrome profile or disable message preview feature?

**Reply by WorldLanguages** [reply]
@warzer13 Why would you get IP banned for looking at your messages? Why not just use a VPN service to hide your real IP address from all software programs?

**Reply by warzer13** [reply]
What I mean is looking at messages on a public internet network (e.g shopping centre free wifi), when you click on the scratch addons icon in bar on the top of google you can see preview (not on scratch). I don't have enough money to afford a private VPN service, public ones will probs get you banned (because if someone else does something bad and make alt ACC then everyone who uses the IP gets banned).

**Reply by mxmou** [reply]
I don't think the Scratch Team blocks accounts that try to use connect from a banned IP address.

Of course, opening the Messaging tab or having "message count on extension icon" enabled will allow Scratch to see your IP address, so those features won't work if you're on a banned network.

**Reply by cobaltt7** [reply]
The Scratch Team (and any website) can access your IP when you have a page open. Using SA does not change this.

**Reply by Chiroyce1** [reply]
Basically how Scratch messages works is by sending a request to the API *(which is a different website than scratch.mit.edu, it's URL is api.scratch.mit.edu, more info on [rest APIs](https://www.redhat.com/en/topics/api/what-is-a-rest-api) here)*. The [/messages](https://scratch.mit.edu/messages) page and extensions you where you give access to Scratch (like SA) can access this API with correct credentials, so it is able to fetch your messages. And yes, it does allow Scratch to view your IP because you are directly making a request to the Scratch API, exactly like how it works in the messages page. There is nothing SA does differently than Scratch in terms of fetching messages.

**Comment by Hans5958** [comment]
> Does it allow scratch to view your IP address when you aren't on the scratch website?

Yes, it does.

> Does the message preview feature risk getting me IP banned 

For what, being in a different country? I don't think they do those bans based on IP in the first place. So, not really.

**Reply by warzer13** [reply]
No, the scratch team bans everyone who is in a IP banned IP address. E.g someone who used there alt to bypass a main ban, etc.

**Reply by Hans5958** [reply]
I have to say I am do not know such think existed on Scratch. I know how IP ban works, but I don't know whether it is implemented on Scratch.

**Reply by Chiroyce1** [reply]
> No, the scratch team bans everyone who is in a IP banned IP address. E.g someone who used there alt to bypass a main ban, etc.

Uh thats just how public IP addresses work. Every client in a WiFi network connected to the internet will have the same public IP (which is actually of the router's)
